index.html

```diff
@@ -260,41 +245,55 @@ <h3>
 service
             </li>
             <li>
-creating the DID document JSON-LD file including a suitable keypair, e.g.,
-using the Koblitz Curve, and storing the <code>did.json</code> file under the
-well-known URL to represent the entire domain, or under the specified path if
-many DIDs will be resolved in this domain.
+creating the DID document JSON, and storing the file as the index of the
+<code>/.well-known/did/</code> directory to represent the entire domain, or as
+the index of the specified path, or as a file with the filename and path matching
+the method specific identifier.
             </li>
           </ol>
 
           <p>
-For example, for the domain name `w3c-ccg.github.io`, the `did.json` will be
-available under the following URL:
+The well-known URL for a domain name is the simplest example:
           </p>
-
-          <pre class="example nohighlight" title="Creating the DID">
+          <pre class="example nohighlight" title="DID at well known path">
 did:web:w3c-ccg.github.io
- -> https://w3c-ccg.github.io/.well-known/did.json
+ -> https://w3c-ccg.github.io/.well-known/did/
```

Comment:
I interpret this as really => `https://w3c-ccg.github.io/.well-known/did/index.json` after some internal server redirect... can you confirm?

Reply:
Index files - the files returned when a directory is requested - are configured on the server.

Index.html has been a default for ages, but any file name may be specified in most webserver configurations.

Reply:
The resolver translates the method specific identifier to a URL and makes an HTTPS request to retrieve the DID document. This changes the HTTPS URL to a directory rather than a full file path in order to meet the requirements in <a href="https://w3c.github.io/did-core/#iana-considerations">IANA Considerations</a> to support multiple representations with appropriate filename extensions and content types (which are mapped to filename extensions in webservers and many clients).

```
->  - https://w3c-ccg.github.io/.well-known/did.json
->  + https://w3c-ccg.github.io/.well-known/did/
```

The first url should always return a content type of `application/json` based on the `.json` filename extension, while the second can return the contents of a file named appropriately for a different DID representation, together with the associated content type HTTP response header.

If we return a file with a `.json` filename extension, and a content type header value associated with some alternate DID representation, we potentially break a long standing best practice. One can argue that `.json` is consistent with `application/did+json` and `application/did+ld+json`, but as soon as we see `application/did+cbor` or some other alternative, we'll no longer have an argument that we're in compliance.